compliance_tool/cli.py

```diff
@@ -1,57 +1,79 @@
+                               help="list available rules for a given profile")
 
 
-def prepare_parser_scan(subparsers) -> None:
-    parser = subparsers.add_parser("scan",
-                                   help="perform a compliance scan")
+def prepare_parser_scan(cli_sub_parsers, global_parser, profile_parser) -> None:
+    parser = cli_sub_parsers.add_parser("scan", parents=[global_parser, profile_parser],
+                                        help="perform a compliance scan")
     parser.set_defaults(func=scan.execute)
-    parser.add_argument("--profile")
-    parser.add_argument("--control")
-    parser.add_argument("--rule")
 
-    parser.add_argument("--html")
-    parser.add_argument("--json")
+    parser.add_argument("--control", metavar="CONTROL_ID",
+                        help="use only control(s) (requirement(s)) with given identifier")
+    parser.add_argument("--rule", metavar="RULE_ID",
+                        help="use only rule(s) with given identifier")
 
+    parser.add_argument("--html", metavar="DESTINATION_FILE",
+                        help="write a human-readable HTML report with the results"
+                             " at the given destination")
+    parser.add_argument("--json", metavar="DESTINATION_FILE",
+                        help="write a JSON-formatted report with the results"
+                             " at the given destination")
 
-def prepare_parser_remediate(subparsers) -> None:
-    parser = subparsers.add_parser("remediate",
-                                   help="perform a system remediation")
+
+def prepare_parser_remediate(cli_sub_parsers, global_parser, profile_parser) -> None:
+    parser = cli_sub_parsers.add_parser("remediate", parents=[global_parser, profile_parser],
+                                        help="perform a system remediation")
     parser.set_defaults(func=remediate.execute)
-    parser.add_argument("--profile")
-    parser.add_argument("--control")
-    parser.add_argument("--rule")
+
+    parser.add_argument("--control", metavar="CONTROL_ID",
+                        help="control (requirement) identifier")
+    parser.add_argument("--rule", metavar="RULE_ID",
+                        help="rule identifier")
 
 
 def prepare_parsers() -> argparse.ArgumentParser:
-    parser = argparse.ArgumentParser()
-    subparsers = parser.add_subparsers(required=True)
-    parser.add_argument("--input",
-                        help="policy definitions (file)")
-    parser.add_argument("--tailoring",
-                        help="policy tailoring customization (file)")
-    prepare_parser_list(subparsers)
-    prepare_parser_scan(subparsers)
-    prepare_parser_remediate(subparsers)
-    return parser
+    global_parser = argparse.ArgumentParser(add_help=False)
+    global_parser.add_argument("--input", metavar="SOURCE_FILE",
+                               help="policy definitions (recognized formats: SCAP Data Stream)")
```

Comment:
It shouldn't depend on a SCAP data stream because we want to leave our door open to other technologies in future. We want the tool to be generic and not depend on SCAP technologies. The design that doesn't involve SCAP will allow us to easily switch to whatever technology we would use in the future.

Reply:
> Do you think that we should somehow hint what format this should be in?

Reply:
Yes, I agree.

Reply:
The option is no longer there.